12.0 lcc members v3 profile improvments

njeudy opened the pull request:
- link partner to existing user.
- Create public profile for Company

njeudy committed:
[NEW] add link partner -> user:
- new odoo_user_id field is computed

njeudy committed:
[NEW] manage public profile for main partner

njeudy committed:
[FIX] update create_public_profile to work for many partners

njeudy committed:
[NEW] add right access to company profile edition

njeudy committed:
[NEW] permit edit main company profile in portal

stephansainleger commented on lcc_members/views/res_partner_view.xml:
```diff
@@ -45,8 +45,8 @@
-                        <label for="user_id" string="User:" />
-                        <field name="user_ids" readonly="1" />
+                        <label for="odoo_user_id" string="Associated User:" />
+                        <field name="odoo_user_id" readonly="1" />
```
Issue on the form view display:
- reading mode: https://drive.lokavaluto.fr/s/9CgF6M2s7A8Nfan
- edit mode: https://drive.lokavaluto.fr/s/iX8azQCrYasdaD6
I think the whole xpath block must be re-designed.

stephansainleger commented on lcc_members/models/res_partner.py:
```diff
@@ -7,6 +7,32 @@
+    "phone_pro",
+    "opening_time",
+    "discount",
+]
```
Missing "is_company" for companies public profiles?

stephansainleger commented on lcc_members/models/res_partner.py:
```diff
@@ -190,6 +236,40 @@ def _membership_state(self):
+            "phone_pro": self.phone_pro,
+            "opening_time": self.opening_time,
+            "discount": self.discount,
+            "is_company": 1,
```
Why is_company = True ? What about the public profile for a person? According to the code this function is called whatever the partner type is

stephansainleger commented on lcc_members/views/res_partner_view.xml:
```diff
@@ -1,6 +1,140 @@
 <?xml version="1.0"?>
 <odoo>
     <data>
+        
```
I do not understand why you create this view, and remove these fields from the partner main form view. Now we cannot edit them directly, we have to go through the main profile and click on the Public Profile at the top of the page.

stephansainleger commented on lcc_members/__init__.py:
```diff
@@ -1,6 +1,17 @@
+from odoo.api import Environment, SUPERUSER_ID
+
+
+def post_init_hook(cr, registry):
```
I am not sure it is a post-init_hook that is needed. At module installation time, no partner is "Main partner" as this field is added with the module update. Then the research can't match any partner, can it?
What about a planned action? or a button to activate the public profile creation process?

stephansainleger commented on lcc_members/views/res_partner_view.xml:
```diff
@@ -220,18 +284,10 @@
                     <page name="social" string="Social networks">
                             <field name="extra_contact_ids" widget="contactInfoWidget" nolabel="True" colspan="1" />
                     </page>
-                    <!-- page Structure access rights -->
```
Why do you also remove these fields from the view?

Commit:
[UPD] Hide Other Positions if partner has a main partner attached

Commit:
[UPD] Public profile can't be customer or supplier

Commit:
[UPD] Design modification of the res.partner form profile data

Commit:
[UPD] Add a button to create public profile from a main profile

Commit:
[UPD] Transfer is_company value when public profile is created

Transfer the main partner's is_company value when the public profile
is created.

Commit:
[UPD] Refactoring methode create_public_profile

Commit:
[UPD] Move structure access right checkboxes in profile header

Commit:
[NEW] Add inactive scheduled action to create public profiles

Target the main profiles without public profiles attached, and attach
the public profile if already existing, or create one.

njeudy committed:
[FIX] affilation company field on crm:
- is required is lead-type = affilation request
- can not quick create / create edit

Commit:
[UPD] partner_profile and access right fields in Contact&Adress form

njeudy committed:
[FIX] other position are only partner_profile_position

Commit:
[UPD] Page Other Positions invisible for all companies

Commit:
[UPD] Add public profile existence check before creating it

Commit:
[UPD] Remove duplicated processes on public partner creation

njeudy committed:
[NEW] default profile is Position

njeudy committed:
[NEW] improve contact and address form on company contact

Commit:
[UPD] Field parent_id is readonly in res partner form
